packages/account/src/BaseSmartAccount.ts

```diff
@@ -270,6 +268,9 @@ export abstract class BaseSmartAccount implements IBaseSmartAccount {
         finalUserOp.callGasLimit = callGasLimit ?? userOp.callGasLimit;
         finalUserOp.preVerificationGas = preVerificationGas ?? userOp.preVerificationGas;
         finalUserOp.paymasterAndData = paymasterAndData ?? userOp.paymasterAndData;
+        if (paymasterServiceData?.mode === PaymasterMode.ERC20) {
```

Comment:
Didn't understand this check? If mode is ERC20 why we setting paymasterAndData to be 0x?

Reply:
Because the ideal flow for token paymaster is like this. The token paymaster expects fee token which we get later and then again call the `buildTokenPaymasterUserOp`

- call bundler for max value
- call paymaster for gas value and paymasterAndData (of sponsored)
	- set paymasterAndData to 0x (as it is needed to be override later)
- call paymaster for fee quote with partialUserOp
- call paymaster for gas values and paymasterAndData (of token)
- call bundler to send tx